src/Symfony/Component/Routing/Loader/ControllerAwareXmlFileLoader.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader;
+
+use Symfony\Component\Config\FileLocatorInterface;
+
+/**
+ * Adds support for a "controller" attribute.
+ *
+ * @author Oleg Voronkovich <oleg-voronkovich@yandex.ru>
+ */
+class ControllerAwareXmlFileLoader extends XmlFileLoader
```

Comment:
However, `RouteCollectionBuilder::add()` also set the controller and is in the component itself...
Eventually, it's probably not worth it to create separate loaders for this.